dot/state/epoch.go

```diff
@@ -884,12 +1006,68 @@ func (s *EpochState) FinalizeBABENextEpochData(finalizedHeader *types.Header) er
 	for e := range s.nextEpochData {
 		if e <= nextEpoch {
 			delete(s.nextEpochData, e)
+			// remove the epoch data from the database
+			if err = s.deleteEpochDataFromDisk(e); err != nil {
+				return fmt.Errorf("cannot delete next epoch data from the database: %w", err)
+			}
 		}
 	}
 
 	return nil
 }
 
+func (s *EpochState) deleteEpochDataFromDisk(epoch uint64) error {
+	configKeysToDelete, err := getDataKeysFromDisk[types.NextEpochData](s.db, nextEpochDataPrefix, epoch)
+	if err != nil {
+		return fmt.Errorf("cannot get next config data keys from disk: %w", err)
+	}
+	batch := s.db.NewBatch()
+	for _, key := range configKeysToDelete {
+		err = batch.Del([]byte(key))
+		if err != nil {
+			return fmt.Errorf("cannot delete next config data from the database: %w", err)
+		}
+	}
+	if err := batch.Flush(); err != nil {
+		return fmt.Errorf("cannot flush deletion batch: %w", err)
+	}
+	return nil
+}
```

Comment:
hey @ramiroJCB I believe that `deleteEpochDataFromDisk` and `deleteNextConfigDataFromDisk` functions can be merged into a single function that accepts as a parameter the prefix, like:

```go
func (s *EpochState) deleteFromDisk(epoch uint64, prefix []byte) error {
	configKeysToDelete, err := getDataKeysFromDisk[types.NextEpochData](s.db, prefix, epoch)
	if err != nil {
		return fmt.Errorf("cannot get next config data keys from disk: %w", err)
	}
	batch := s.db.NewBatch()
	for _, key := range configKeysToDelete {
		err = batch.Del([]byte(key))
		if err != nil {
			return fmt.Errorf("cannot delete next config data from the database: %w", err)
		}
	}
	if err := batch.Flush(); err != nil {
		return fmt.Errorf("cannot flush deletion batch: %w", err)
	}
	return nil
}

// then you can just use like:
if err = s.deleteFromDisk(e, nextEpochDataPrefix); err != nil {
	return fmt.Errorf("cannot delete next config data from the database: %w", err)
}
//or 
if err = s.deleteFromDisk(e, nextConfigDataPrefix); err != nil {
	return fmt.Errorf("cannot delete next config data from the database: %w", err)
}
```